cranelift/codegen/src/isa/x64/inst/args.rs

```diff
@@ -313,14 +313,16 @@ impl Amode {
     ) {
         match self {
             Amode::ImmReg { base, .. } => {
-                collector.reg_use(*base);
+                if *base != regs::rbp() && *base != regs::rsp() {
```

Comment:
I think we need to mirror this conditional over to `with_allocs` [here](https://github.com/bytecodealliance/wasmtime/blob/c77bec4dcb33973a386d924474d95bab6b23e263/cranelift/codegen/src/isa/x64/inst/args.rs#L370) as well; otherwise we have a thing that consumes more allocs than the operands it reports. This will be kind-of-harmless today because `Amode`'s operands are always placed at the end of the inst by convention (because they're variable-length and that was easier wrt reuse-indices) but if a pseudoinst ever grows two `Amode`s or puts something after an `Amode` that could be an issue.